types/streamx/index.d.ts

```diff
@@ -0,0 +1,309 @@
+    TEvents extends StreamEvents = StreamEvents
+> extends EventEmitter {
+    constructor(opts?: StreamOptions<Stream<TByteType>, TByteType>);
+    _open(cb: Callback): void;
+    _destroy(cb: Callback): void;
+    _predestroy(cb: Callback): void;
+    readonly readable: TReadable;
+    readonly writable: TWritable;
+    readonly destroyed: boolean;
+    readonly destroying: boolean;
+    destroy(error?: Error | null): void;
+
+    addListener<TEvent extends EventName<TEvents>>(event: TEvent, listener: EventListener<TEvents, TEvent>): this;
+    on<TEvent extends EventName<TEvents>>(event: TEvent, listener: EventListener<TEvents, TEvent>): this;
+    once<TEvent extends EventName<TEvents>>(event: TEvent, listener: EventListener<TEvents, TEvent>): this;
+    removeListener<TEvent extends EventName<TEvents>>(event: TEvent, listener: EventListener<TEvents, TEvent>): this;
+    off<TEvent extends EventName<TEvents>>(event: TEvent, listener: EventListener<TEvents, TEvent>): this;
+    removeAllListeners(event?: EventName<TEvents>): this;
+    setMaxListeners(n: number): this;
+    getMaxListeners(): number;
+    listeners<TEvent extends EventName<TEvents>>(event: TEvent): Array<EventListener<TEvents, TEvent>>;
+    rawListeners<TEvent extends EventName<TEvents>>(event: TEvent): Array<EventListener<TEvents, TEvent>>;
+    emit<TEvent extends EventName<TEvents>>(
+        event: TEvent,
+        ...rest: Parameters<EventListener<TEvents, TEvent>>
+    ): boolean;
+    listenerCount(event: EventName<TEvents>): number;
+
+    // Added in Node 6...
+    prependListener<TEvent extends EventName<TEvents>>(event: TEvent, listener: EventListener<TEvents, TEvent>): this;
+    prependOnceListener<TEvent extends EventName<TEvents>>(
+        event: TEvent,
+        listener: EventListener<TEvents, TEvent>,
+    ): this;
+    eventNames(): Array<string | symbol>;
+}
+
+export type MapFunction<TThis, TIn, TOut> = ((this: TThis, input: TIn) => TOut) | null | undefined;
+export type ByteLengthFunction<TThis, TData> = ((this: TThis, data: TData) => number) | null | undefined;
+
+export interface BaseReadableOptions<
+    TStream extends Readable<TType, TMapType, TByteType, any, any>,
+    TType,
+    TMapType,
+    TByteType
+> extends StreamOptions<TStream, TByteType> {
+    read?: (this: TStream, cb: ResultCallback<TType>) => void;
+    byteLengthReadable?: ByteLengthFunction<TStream, TByteType>;
+}
+
+export type ReadableOptions<
+    TStream extends Readable<TType, TMapType, TByteType, any, any>,
+    TType,
+    TMapType,
+    TByteType,
+    TMapFallback = any
+> = BaseReadableOptions<TStream, TType, TMapType, TByteType> &
+    (
+        | {}
+        | {
+              map?: TMapFallback;
+              mapReadable: MapFunction<TStream, TType, TMapType>;
+          }
+        | {
+              map: MapFunction<TStream, TType, TMapType>;
+          }
+    );
+
+/* tslint:disable-next-line strict-export-declare-modifiers */
```

Comment:
Why is this eslint exception necessary?

Reply:
`FromType` is a utility type for which there doesn't seem to be any use outside of this package, which is why I refrained from exporting it. All types that are not exported cause a lint error.

Reply:
@amcasey The easier way would be to "just export" `FromType` but I consciously decided not to export `FromType` as it looks to be utterly unhelpful outside of this package _(its an internal helper to identify the result of `.from`)_. If that is an unreasonable stance it can be exported as well.

Reply:
I was not suggesting exporting this type, but rather adding a specific statement, `export {};`, which tells TypeScript not to automatically export every declaration.

Reply:
I didn't know this is a thing. I pushed https://github.com/DefinitelyTyped/DefinitelyTyped/pull/49810/commits/451906b3f17b6d16397e1552af5ea564c0283c32 to fix this issue. Thank you for pointing it out!